src/main/webapp/app/shared/layout/header/canvas-header.tsx

```diff
@@ -0,0 +1,173 @@
+import './header.scss';
+import React, { useEffect, useState } from 'react';
+import { ActionButton, View, DialogContainer, TooltipTrigger, Tooltip, MenuTrigger, Menu, Item, Picker } from '@adobe/react-spectrum';
+import MoreSmallListVert from '@spectrum-icons/workflow/MoreSmallListVert';
+import SaveAsFloppy from '@spectrum-icons/workflow/SaveAsFloppy';
+import GraphBarVerticalAdd from '@spectrum-icons/workflow/GraphBarVerticalAdd';
+import CollectionEdit from '@spectrum-icons/workflow/CollectionEdit';
+import VisualizationsList from 'app/entities/visualizations/visualizations-list';
+import { IRootState } from 'app/shared/reducers';
+import { connect } from 'react-redux';
+import {
+  createEntity as addVisualmetadataEntity,
+  deleteEntity as deleteVisualmetadataEntity,
+} from 'app/entities/visualmetadata/visualmetadata.reducer';
+import { toggleEditMode } from 'app/shared/reducers/application-profile';
+import { saveViewState } from 'app/entities/views/views.reducer';
+import { getEntities as getDashboardEntities } from 'app/entities/dashboard/dashboard.reducer';
+import { getAllEntities as getViewEntities } from 'app/entities/views/views.reducer';
+import { RouteComponentProps, useHistory } from 'react-router-dom';
+
+interface ICanvasHeaderProps extends StateProps, DispatchProps, RouteComponentProps<{ id: string; viewId: string }> {}
+const CanvasHeader = (props: ICanvasHeaderProps) => {
+  const history = useHistory();
+  const [isVisualizationsModelOpen, setVisualizationsModelOpen] = useState(false);
+  const [dashboardName, setDashboardName] = useState('');
+  const [viewName, setViewName] = useState('');
+
+  const changeViewAndUpdateDashboard = id => {
+    setViewName(id);
+    history.push(`/dashboards/${dashboardName}/${id}/build`);
+    window.location.reload();
```

Comment:
why are you reloading the page?

Reply:
when you change view then load the selected view